File: C/IslandPerimeter.c

int islandPerimeter(int **grid, int gridSize, int *gridColSize)
{
	int M = gridSize, N = *gridColSize, islands = 0, neighbours = 0;
	for (int i = 0; i < M; ++i)
	{
		for (int j = 0; j < N; ++j)
		{
			if (grid[i][j] == 1)
			{
				++islands; // count islands
				if (i < M - 1 && grid[i + 1][j] == 1)
					++neighbours; // count down neighbours
				if (j < N - 1 && grid[i][j + 1] == 1)
					++neighbours; // count right neighbours
			}
		}
	}
	return islands * 4 - neighbours * 2;
}

File: python/IslandPerimeter.py
from typing import List


class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        M, N = len(grid), len(grid[0])
        islands, neighbours = 0, 0
        for i in range(M):
            for j in range(N):
                if grid[i][j] == 1:
                    islands += 1
                    if i < M - 1 and grid[i + 1][j] == 1:
                        neighbours += 1
                    if j < N - 1 and grid[i][j + 1] == 1:
                        neighbours += 1
        return islands * 4 - neighbours * 2
